### PR TITLE
[local testing] Track executed migrations when locally testing

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- suggestion for tracking clickhouse migrations on tests so there's no need to migrate on running every single test
- test setup takes 3 seconds instead of 12 when the tables are migrated
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Comment by EDsCODE** [comment]
This code is gross right now but wanted to test it and give you a look @hazzadous

**Comment by hazzadous** [comment]
mmm much nicer:

```
(env) ➜  posthog git:(speed-up-local-ee-tests) ✗ pytest  ee/clickhouse/queries/funnels/test/test_funnel_correlation.py -k  test_no_divide_by_zero_errors
==================================================== test session starts =====================================================
platform darwin -- Python 3.9.4, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
django: settings: posthog.settings (from ini)
rootdir: /Users/harry/src/posthog/posthog, configfile: pytest.ini
plugins: cov-2.12.1, syrupy-1.4.6, split-0.3.3, celery-4.4.2, xdist-2.4.0, django-4.1.0, forked-1.3.0, mock-3.5.1
collected 16 items / 15 deselected / 1 selected                                                                              

ee/clickhouse/queries/funnels/test/test_funnel_correlation.py .                                                        [100%]

-------------------------------------------------- snapshot report summary ---------------------------------------------------

============================================== 1 passed, 15 deselected in 1.38s ==============================================
```

I'll have a look through the code.

btw I played with not dropping the tables and just truncating at the beginning, but truncating is very slow 🤷

**Comment by hazzadous** [comment]
How about we just simplify this and say,

```
        if not django_db_keepdb or not database.db_exists:
            database.create_database()
            create_clickhouse_tables()
```

i.e. we only create tables on initial create_database AND we also don't bother cleaning out the tables (which I think is the affect of this PR but I'm thinking we don't need to track anything. So this would be in line with https://pytest-django.readthedocs.io/en/latest/database.html#reuse-db-reuse-the-testing-database-between-test-runs except we don't truncate, but presumably that's not an issue as we're already not truncating between test runs as the fixture is scoped to the package

File: ee/clickhouse/sql/cohort.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
     cluster=CLICKHOUSE_CLUSTER, engine=table_engine("cohortpeople", "sign", COLLAPSING_MERGE_TREE), storage_policy=""
 )
 
-DROP_COHORTPEOPLE_TABLE_SQL = f"DROP TABLE IF EXISTS cohortpeople ON CLUSTER {CLICKHOUSE_CLUSTER}"
+DROP_COHORTPEOPLE_TABLE_SQL = f"TRUNCATE TABLE IF EXISTS cohortpeople ON CLUSTER {CLICKHOUSE_CLUSTER}"
```
Comment (macobo) [inline ee/clickhouse/sql/cohort.py:23]: Note the name is now misleading

File: ee/conftest.py [hunk 2]
```diff
@@ -80,7 +88,25 @@ def django_db_setup(django_db_setup, django_db_keepdb):
         if not django_db_keepdb or not database.db_exists:
             database.create_database()
 
-        reset_clickhouse_tables()
+        num_migrated_tables = 0
+
+        with django_db_blocker.unblock():
```
Comment (macobo) [inline ee/conftest.py:93]: Nit: extract fn, this would need a name:)

**Comment by EDsCODE** [comment]
Simplified it along the lines of what you described. No need for extra tracking now. Speed benefit should be the same. @hazzadous

**Review by hazzadous — APPROVED** [review]
Cool, could probably do with some documentation and a notification of the change in slack or something, just in case people start having issues?

**Pull request merged**